### ISSUE TITLE
Cannot read properties of undefined (reading 'workflows.argoproj.io/completed')

### ISSUE THREAD

**Issue by mozarik** [argoproj/argo-workflows#11944]
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issues exists when I tested with `:latest`
- [ ] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/master/docs/CONTRIBUTING.md))

### What happened/what you expected to happen?

# What happend

it showed me this 
![image](https://github.com/argoproj/argo-workflows/assets/39885486/6a50bf32-b226-4ede-af71-f9e86e3bc325)

# What expected to happen
The UI refreshed if new workflow is running

### Version

v3.5.0-rc1

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

1. have this in your workflow-controller-configmaps.yaml
```
  columns: |
    # Adds a column to the Workflow List View
    - # The name of this column, e.g., "Workflow Completed".
      name: Workflow Completed
      # The type of this column, "label" or "annotation".
      type: label
      # The key of the label or annotation, e.g., "workflows.argoproj.io/completed".
      key: workflows.argoproj.io/completed
```

2. go to https://<redacted>/workflows/argo

3. submit workflow by `kubectl -n argo create -f workflow.yaml` 

4. It will show what happened

### Logs from the workflow controller

```text
time="2023-10-04T18:06:04.716Z" level=info msg="Processing workflow" namespace=argo workflow=dag-inline-2mxjb
time="2023-10-04T18:06:04.732Z" level=info msg="Updated phase  -> Running" namespace=argo workflow=dag-inline-2mxjb
time="2023-10-04T18:06:04.733Z" level=warning msg="Node was nil, will be initialized as type Skipped" namespace=argo workflow=dag-inline-2mxjb
time="2023-10-04T18:06:04.733Z" level=info msg="was unable to obtain node for , letting display name to be nodeName" namespace=argo workflow=dag-inline-2mxjb
time="2023-10-04T18:06:04.733Z" level=info msg="DAG node dag-inline-2mxjb initialized Running" namespace=argo workflow=dag-inline-2mxjb
time="2023-10-04T18:06:04.733Z" level=warning msg="was unable to obtain the node for dag-inline-2mxjb-2717731684, taskName a"
time="2023-10-04T18:06:04.733Z" level=warning msg="was unable to obtain the node for dag-inline-2mxjb-2717731684, taskName a"
time="2023-10-04T18:06:04.733Z" level=info msg="All of node dag-inline-2mxjb.a dependencies [] completed" namespace=argo workflow=dag-inline-2mxjb
time="2023-10-04T18:06:04.733Z" level=warning msg="Node was nil, will be initialized as type Skipped" namespace=argo workflow=dag-inline-2mxjb
time="2023-10-04T18:06:04.733Z" level=info msg="Pod node dag-inline-2mxjb-2717731684 initialized Pending" namespace=argo workflow=dag-inline-2mxjb
time="2023-10-04T18:06:04.752Z" level=info msg="Created pod: dag-inline-2mxjb.a (dag-inline-2mxjb--2717731684)" namespace=argo workflow=dag-inline-2mxjb
time="2023-10-04T18:06:04.752Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=dag-inline-2mxjb
time="2023-10-04T18:06:04.752Z" level=info msg=reconcileAgentPod namespace=argo workflow=dag-inline-2mxjb
time="2023-10-04T18:06:04.766Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=1812002943 workflow=dag-inline-2mxjb
time="2023-10-04T18:06:14.718Z" level=info msg="Processing workflow" namespace=argo workflow=dag-inline-2mxjb
time="2023-10-04T18:06:14.718Z" level=info msg="Task-result reconciliation" namespace=argo numObjs=0 workflow=dag-inline-2mxjb
time="2023-10-04T18:06:14.718Z" level=info msg="node changed" namespace=argo new.message= new.phase=Succeeded new.progress=0/1 nodeID=dag-inline-2mxjb-2717731684 old.message= old.phase=Pending old.progress=0/1 workflow=dag-inline-2mxjb
time="2023-10-04T18:06:14.718Z" level=info msg="Outbound nodes of dag-inline-2mxjb set to [dag-inline-2mxjb-2717731684]" namespace=argo workflow=dag-inline-2mxjb
time="2023-10-04T18:06:14.718Z" level=info msg="node dag-inline-2mxjb phase Running -> Succeeded" namespace=argo workflow=dag-inline-2mxjb
time="2023-10-04T18:06:14.718Z" level=info msg="node dag-inline-2mxjb finished: 2023-10-04 18:06:14.718514255 +0000 UTC" namespace=argo workflow=dag-inline-2mxjb
time="2023-10-04T18:06:14.718Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=dag-inline-2mxjb
time="2023-10-04T18:06:14.718Z" level=info msg=reconcileAgentPod namespace=argo workflow=dag-inline-2mxjb
time="2023-10-04T18:06:14.718Z" level=info msg="Updated phase Running -> Succeeded" namespace=argo workflow=dag-inline-2mxjb
time="2023-10-04T18:06:14.718Z" level=info msg="Marking workflow completed" namespace=argo workflow=dag-inline-2mxjb
time="2023-10-04T18:06:14.724Z" level=info msg="cleaning up pod" action=deletePod key=argo/dag-inline-2mxjb-1340600742-agent/deletePod
time="2023-10-04T18:06:14.730Z" level=info msg="Workflow update successful" namespace=argo phase=Succeeded resourceVersion=1812003354 workflow=dag-inline-2mxjb
time="2023-10-04T18:06:14.730Z" level=info msg="Queueing Succeeded workflow argo/dag-inline-2mxjb for delete in 240h0m0s due to TTL"
time="2023-10-04T18:06:14.739Z" level=info msg="cleaning up pod" action=labelPodCompleted key=argo/dag-inline-2mxjb--2717731684/labelPodCompleted
time="2023-10-04T18:08:08.336Z" level=info msg="Queueing Succeeded workflow argo/dag-inline-2mxjb for delete in 239h58m6s due to TTL"
time="2023-10-04T18:13:26.331Z" level=info msg="Queueing Succeeded workflow argo/dag-inline-2mxjb for delete in 239h52m48s due to TTL"
```

### Logs from in your workflow's wait container

```text
Unrelated
```

**Comment by terrytangyuan** [comment]
Can you paste the logs from your browser developer console?

**Comment by terrytangyuan** [comment]
The formatting of your `columns` is incorrect. You have comments after the hyphen. Try this:

```
  columns: |
    - name: Workflow Completed
      type: label
      key: workflows.argoproj.io/completed
```

**Comment by terrytangyuan** [comment]
I don't think you've pasted the whole logs from workflow controller. It should be error message indicating the configmap is malformed or something.

**Comment by agilgur5** [comment]
@terrytangyuan the hyphens are actually from the example in the docs. In fact, they're from my change that added `columns` to the example: https://github.com/argoproj/argo-workflows/pull/11297 .

It is valid YAML to have comments in between keys of a map. but possibly the ConfigMap string -> YAML parsing if causing an error?

**Comment by agilgur5** [comment]
Nope the parsing works correctly, so the ConfigMap is fine. Just tested it locally and modified my ConfigMap to have the same comments.

I think this is actually the bug that #11638 fixed -- when a Workflow has no labels (i.e. `labels` is `undefined`), then you'll get this error (it's trying to access this label on an `undefined` variable).

> v3.5.0-rc1

That fix should've made it into 3.5.0-rc1 one though

**Comment by agilgur5** [comment]
> That fix should've made it into 3.5.0-rc1 though

Oh nvm, #11638 missed the rc1 cut-off by a few days.

@mozarik this should be fixed in [3.5.0-rc2](https://github.com/argoproj/argo-workflows/blob/c04873e7e33e00e95c4cad4eace6b517ea1c5aa7/CHANGELOG.md#v350-rc2-2023-09-20)

**Comment by mozarik** [comment]
Sry i got sleepy. Thank you @agilgur5
